subscription.go

```diff
@@ -5,66 +5,161 @@ import (
 	"fmt"
 
 	"github.com/graphql-go/graphql/gqlerrors"
-	"github.com/graphql-go/graphql/language/ast"
+	"github.com/graphql-go/graphql/language/parser"
+	"github.com/graphql-go/graphql/language/source"
 )
 
 // SubscribeParams parameters for subscribing
 type SubscribeParams struct {
-	Schema          Schema
-	Document        *ast.Document
-	RootValue       interface{}
-	ContextValue    context.Context
+	Schema        Schema
+	RequestString string
+	RootValue     interface{}
+	// ContextValue    context.Context
 	VariableValues  map[string]interface{}
 	OperationName   string
 	FieldResolver   FieldResolveFn
 	FieldSubscriber FieldResolveFn
 }
 
-// Subscribe performs a subscribe operation
-func Subscribe(ctx context.Context, p SubscribeParams) chan *Result {
-	resultChannel := make(chan *Result)
+// SubscriptableSchema implements `graphql-transport-ws` `GraphQLService` interface: https://github.com/graph-gophers/graphql-transport-ws/blob/40c0484322990a129cac2f2d2763c3315230280c/graphqlws/internal/connection/connection.go#L53
+// you can pass `SubscriptableSchema` to `graphql-transport-ws` `NewHandlerFunc`
+type SubscriptableSchema struct {
```

Comment:
While I agree this is useful, there is no reason this needs to be part of the library. It can be implemented outside of the library as all the types it relies on are exported.

Reply:
Where should this type live? handler package?

Reply:
i removed it, this will probably live in the handler package as a `NewSubscriptionHandler` function that uses `graphql-transport-ws` under the hood

If you want to try some subscirption i made an example [here](https://github.com/remorses/graphql-go-subscription-example)

Reply:
> i removed it, this will probably live in the handler package as a `NewSubscriptionHandler` function that uses `graphql-transport-ws` under the hood
> 
> If you want to try some subscirption i made an example [here](https://github.com/remorses/graphql-go-subscription-example)

yes, a different package. Feel free to PR it into the graphql-go-tools if you don't have another place for it. My request there is that it be its own package in a subfolder